ssh.c

```diff
@@ -257,6 +258,31 @@ resolve_host(const char *name, int port, int logerr, char *cname, size_t clen)
 	int gaierr;
 	LogLevel loglevel = SYSLOG_LEVEL_DEBUG1;
 
+	if (name[0] == '/') {
+		struct sockaddr_un *sunaddr;
+
+		if (strlen(name) > sizeof(sunaddr->sun_path)) {
```

Comment:
Oh, good catch! That needs fixing in `channels.c` / `connect_to_helper()` too then - I copied from there. I guess that needs a separate PR. Do you want to create one for that?

Reply:
Yep! It wouldn't be very BSD-ish to go back to use strncpy to win one more character in the path. New PR fixing all occurrances at https://github.com/openssh/openssh-portable/pull/436 .

Reply:
It's actually worse than that, and has nothing to do with `strlcpy()`: `sun_path` needs to be terminated with a `NUL` character, so that has to be counted. If not, anything trying to access it as a string would run into the memory that follows.

Reply:
Fixed.